addons/account/models/account_tax.py

```diff
@@ -608,6 +608,16 @@ def recompute_base(base_amount, fixed_amount, percent_amount, division_amount):
                         total_included_checkpoints[i] = base
                         store_included_tax_total = False
                 i -= 1
+            # In case we're computing the total_excluded from a value that includes the taxes, we should check if an
+            # early pay discount (must be mixed) was applied on that value. If so, we apply the discount to the tax.
+            # In one example: let's say: base = 100, discount = 2%, tax = 21%
+            # the total will be calculated as: total = base + (base * (1 - discount)) * tax
+            # If we manipulate the equation to get the base from the total, we'll have base = total / ((1 - discount) * tax + 1)
+            # In the recompute_base function, it's missing the (1-discount) multiplication with the tax so we apply it here.
+            # This only handles the case of a single percentage tax.
+            if self._context.get('early_pay_discount_mixed', False):
+                discount_percentage = self._context.get('discount_percentage')
+                incl_percent_amount *= (1.0 - discount_percentage / 100.0)
```

Comment:
Btw, making a `get` before using it is not helping.
If `self._context.get('discount_percentage')` is returning None, you will get a traceback on `incl_percent_amount *= (1.0 - discount_percentage / 100.0)`. So, in that case, it's better to just write `self._context['discount_percentage']` or add an additional if checking if `discount_percentage` is `None` or not

Reply:
@alialfie Because you can put everything on the context and it's not a good practice. We could also remove all method parameters and only rely on context !
Also, since behavior is needed only for a single case in a single model with a specific config, It's better to keep it isolated as much as possible